src/channel.ts

```diff
@@ -39,16 +42,16 @@ export type EventMatcher = string | string[] | RegExp | Predicate | boolean;
  * stream. To do that see `query`.
  * @see query
  */
-export interface Filter {
-  (item: Event): any;
+export interface Filter<T> {
+  (item: T): any;
 }
 
 /**
  * The function you assign to `.on(eventType, fn)`
  * events is known as a Listener. It receives the event.
  */
-export interface Listener {
-  (item: Event): any;
+export interface Listener<T> {
+  (item: T): any;
```

Comment:
I'd more likely expect the return of a listener to either be void, or a DestroyListener<T> method

Reply:
The Subscription encapsulates the destroy method, but takes care of its idempotence.

Reply:
When a listener is invoked in response to an event, it ideally returns an Observable, which contains its own unsubscribe logic for destroying itself. 

But for compatibility reasons, Promises, thunks, and plain values are allowed as well. The return value is often significant, because if you return a Promise or Observable, and the next call to the handler is concurrent with the previous, the listener can be set up to enqueue easily.

This listener setup (for which I don't have a good name, nor type, its only interface is Subscription) will not download more than one `bigassfile.json` at a time..

```
listen('bigfile/download', ({ payload: { id } } => fetch('bigassfile.json'), { mode: 'serial' })
```

The idea here is that the mode is like switching between `takeEvery` `takeLatest` and `takeLeading` in redux-saga, except it's more declarative and controlled separately from the underlying logic of handling the event.

Reply:
Incidentally @rplotkin, the [next PR](https://github.com/deanius/polyrhythm/pull/20) brings in generator support to make polyrhythm even interchangable with Redux Saga (but without the Redux dependency). I've seen a place or two this may make Redux Saga code simpler - more PRs to come are possible.